statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,480 @@
+        """
+        The __str__ method for our data
+        """
+        print("Number of observations:", len(self.strat))
+        print("Sum of weights:", self.weights.sum())
+        print("Number of strata:", self.nstrat)
+        print("The number of clusters per stratum:", self.ncs)
+
+
+    def _check_args(self, strata, cluster, weights):
+        """
+        Minor error checking to make sure user supplied any of 
+        strata, cluster, or weights. For unspecified subgroup labels
+        an array of ones is created
+
+        Parameters
+        ----------
+        strata : array-like or None                                                                                                                                 
+            Strata for each observation. If none, an array 
+            of ones is constructed                                                                                                                                 
+        cluster : array-like or None                                                                                                                 
+            Cluster for each observation. If none, an array 
+            of ones is constructed                                                                                                         
+        weights : array-like or None                                                                                                                               
+            The weight for each observation. If none, an array 
+            of ones is constructed
+
+        Returns
+        -------
+        vals : list
+            List of strata, cluster, and weights
+        """
+        if all([x is None for x in (strata, cluster, weights)]):
+            raise ValueError("At least one of strata, cluster, and weights musts not be None")
+        v = [len(x) for x in (strata, cluster, weights) if x is not None]
+        if len(set(v)) != 1:
+            raise ValueError("lengths of strata, cluster, and weights are not compatible")
+        n = v[0]
+        vals = []
+        for x in (strata, cluster, weights):
+            if x is None:
+                vals.append(np.ones(n))
+            else:
+                vals.append(np.asarray(x))
+
+        return vals[0], vals[1], vals[2] 
+
+
+class SurveyStat(SurveyDesign):
+    """                                                                                                                                              
+    Holds methods to calculate standard error of statistics.                                                                                                 
+                                                                                                                                                     
+    Parameters                                                                                                                                          
```

Comment:
@kshedden That's fine her we only use class docstring and the `__init__` method doesn't get a docstring. (In general or by default we don't list `__xxx__` methods in the documentation.)